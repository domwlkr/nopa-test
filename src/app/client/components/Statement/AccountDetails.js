import React, {PropTypes} from 'react';

const AccountDetails = (props) => {
  return (
    <div className="account-details">
      <div className="account-details__left">
        <span>{props.bank.name}</span>
        <span>{props.accountDetails.surname}</span>
      </div>
      <div className="account-details__right">
        <span>Current account</span>
        <span>{props.accountDetails.accountNumber}</span>
        <span>{props.accountDetails.sortCode}</span>
      </div>
    </div>
  );
};

AccountDetails.propTypes = {
  accountDetails: PropTypes.object,
  bank: PropTypes.object
};

export default AccountDetails;