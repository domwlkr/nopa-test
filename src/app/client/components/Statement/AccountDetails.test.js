import React from 'react';
import AccountDetails from './AccountDetails';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

const bank = {name: 'Barclays', logo: 'test-file-stub'};
const details = {surname: 'Walker', accountNumber: '90909090', sortCode: '220099'};

test('Renders correctly', () => {
  const accountDetails = renderer.create(
    <AccountDetails bank={bank} accountDetails={details} />
  ).toJSON();

  expect(accountDetails).toMatchSnapshot();
});


test('Displays the correct bank', () => {
  const accountDetails = shallow(
    <AccountDetails bank={bank} accountDetails={details} />
  );

  expect(accountDetails.instance().props.bank.name).toEqual('Barclays');
});


test('Displays the correct account number', () => {
  const accountDetails = shallow(
    <AccountDetails bank={bank} accountDetails={details} />
  );

  expect(accountDetails.instance().props.accountDetails.accountNumber).toEqual('90909090');
});


test('Displays the correct sort code', () => {
  const accountDetails = shallow(
    <AccountDetails bank={bank} accountDetails={details} />
  );

  expect(accountDetails.instance().props.accountDetails.sortCode).toEqual('220099');
});


test('Displays the surname', () => {
  const accountDetails = shallow(
    <AccountDetails bank={bank} accountDetails={details} />
  );

  expect(accountDetails.instance().props.accountDetails.surname).toEqual('Walker');
});