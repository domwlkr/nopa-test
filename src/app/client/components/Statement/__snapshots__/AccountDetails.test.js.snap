// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="account-details"
>
  <div
    className="account-details__left"
  >
    <span>
      Barclays
    </span>
    <span>
      Walker
    </span>
  </div>
  <div
    className="account-details__right"
  >
    <span>
      Current account
    </span>
    <span>
      90909090
    </span>
    <span>
      220099
    </span>
  </div>
</div>
`;
